distributions/openhab/src/main/resources/bin/update.ps1

```diff
@@ -69,190 +12,582 @@ Function Update-openHAB {
     The version to upgrade to.
     .PARAMETER Snapshot
     Upgrade to a snapshot version ($true) or a release version ($false) (default: $false)
+    .PARAMETER AutoConfirm
+    Automatically confirm update (used for headless mode)
     .PARAMETER SkipNew
     Internal use only. For skipping the check for a new update.ps1
     .EXAMPLE
     Update the openHAB distribution in the current directory to the current stable version
     Update-openHAB
     .EXAMPLE
     Update the openHAB distribution in the C:\oh-snapshot directory to the next snapshot version
-    Update-openHAB -OHDirectory C:\oh-snapshot -OHVersion 2.2.0 -Snapshot $true
-    #>
+    Update-openHAB -OHDirectory C:\oh-snapshot -OHVersion 2.3.0 -Snapshot $true
+#>
 
+Function Update-openHAB() {
     [CmdletBinding()]
     param(
-        [Parameter(ValueFromPipeline=$True)]
+        [Parameter(ValueFromPipeline = $True)]
         [string]$OHDirectory = ".",
-        [Parameter(ValueFromPipeline=$True)]
-        [string]$OHVersion = "2.1.0",
-        [Parameter(ValueFromPipeline=$True)]
+        [Parameter(ValueFromPipeline = $True)]
+        [string]$OHVersion,
+        [Parameter(ValueFromPipeline = $True)]
         [boolean]$Snapshot = $false,
-        [Parameter(ValueFromPipeline=$True)]
-        [boolean]$SkipNew = $false
+        [Parameter(ValueFromPipeline = $True)]
+        [boolean]$AutoConfirm = $false,
+        [Parameter(ValueFromPipeline = $True)]
+        [boolean]$SkipNew = $false,
+        [Parameter(ValueFromPipeline = $True)]
+        [boolean]$KeepUpdateScript = $false  # sssh - secret switch ;)
     )
 
-    begin {}
-    process {
+    function DownloadFiles() {
+        param(    
+            [Parameter(Mandatory = $True)]
+            [string] $DownloadSource, 
+            [Parameter(Mandatory = $True)]
+            [string] $OutputFile
+        ) 
+    
+        $uri = New-Object "System.Uri" "$DownloadSource"
+
```

Comment:
I really hate that the fancy percentage code that (yes, I found) is gone...
It was nice to know how long this was going to take.

I've found that I have needed to add the following code to get it to work.
`[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12`

Reply:
That 'fancy percentage code' was always messing up for me (seemed like it was doing new lines on every update or so - made the cursor jump around like crazy as well).  You should still see the progress (ie bytes downloaded) from the native invoke-webrequest progress meter.

Surprised on the tls12 flag - worked fine for me but I've added that anyway

Reply:
Yeah - that's actually an issue I tracked down yesterday (don't have the issue link - but it's basically if you run in the console, the download progress meter slows it down drastically.  If you run in a powershell console, it flies.  I'll look into putting the old meter back in and trying to fix it for my stuff.

BTW - almost ready to commit a new update that is fairly big changes (went ahead and did the environment variables in all scripts - mainly because I'm also doing an installer and was thinking about allowing that via the installer as well)..

Reply:
Nice!

Let me know what you need to for testing the downloading.